tokio/src/runtime/scheduler/multi_thread/worker.rs

```diff
@@ -1090,17 +1082,16 @@ impl Handle {
             true
         } else {
             // Push to the LIFO slot
-            let prev = core.lifo_slot.take();
-            let ret = prev.is_some();
-
-            if let Some(prev) = prev {
-                core.run_queue
-                    .push_back_or_overflow(prev, self, &mut core.stats);
+            match core.run_queue.push_lifo(task) {
+                Some(prev) => {
+                    // There was a previous task in the LIFO slot which needs
+                    // to be pushed to the back of the run queue.
+                    core.run_queue
+                        .push_back_or_overflow(prev, self, &mut core.stats);
+                    true
+                }
+                None => false,
```

Comment:
This boolean indicates whether other worker threads are notified that there is work available to steal. Since the lifo slot is now stealable, it should not be false in this scenario.

Reply:
Is this something we could have caught with a loom test? Maybe put something in the lifo slot and then block the thread, concurrently with another thread doing nothing?